internal/broker/broker.go

```diff
@@ -429,3 +441,37 @@ func (m *mcpBrokerImpl) Close(_ context.Context, downstreamSession downstreamSes
 func (m *mcpBrokerImpl) MCPServer() *server.MCPServer {
 	return m.listeningMCPServer
 }
+
+// Get the authorization header needed for a particular MCP upstream
+func getAuthorizationHeaderForUpstream(upstream *upstreamMCP) string {
+	// We don't store the authorization in the config.yaml, which comes from a ConfigMap.
+	// Instead it is passed to the Broker pod through an env var (typically from a Secret)
+	// The format is
+	// url1=value1+url2=value2+url3=value3
+	// e.g.
+	// UPSTREAM_AUTHORIZATIONS=http://localhost:9090/mcp=Bearer 1234
+	authorizations := os.Getenv("UPSTREAM_AUTHORIZATIONS")
+
+	// Not defined
+	if authorizations == "" {
+		return ""
+	}
+
+	for _, auth := range strings.Split(authorizations, "+") {
+		authParts := strings.SplitN(auth, "=", 2)
+		if len(authParts) != 2 {
+			// It would be better to print the incorrect fragment, but we don't want
+			// to log a possibly important bearer key
+			log.Printf("Invalid auth encoding (%d parts)", len(authParts))
+			continue
+		}
+
+		if authParts[0] == string(upstream.upstreamMCP) {
```

Comment:
I wonder would it make sense to use the MCP name from the config as the key here?